### PR TITLE
Further bobtail validation

### PR DESCRIPTION
**Pull request by bissias**
This PR completes the implementation of all bobtail validation methods that are currently possible given the state of completion of other parts of the code. The notable exception is CheckBobtailBlockHeader, which is waiting on the definition of CBobtailBlockHeader. Additionally, unnecessary bobtail validation methods have been removed.

### CODE REVIEW

File: src/bobtail/bobtail.cpp [hunk 1]
```diff
@@ -98,6 +89,31 @@ bool CheckBobtailPoWFromOrderedProofs(std::vector<arith_uint256> proofs, arith_u
...
+    if (bnTarget > UintToArith256(params.powLimit))
+    {
+        LOG(WB, "Illegal target value bnTarget=%d for pow limit\n", bnTarget.getdouble());
+        return false;
+    }
+
+    arith_uint256 pow = UintToArith256(header.GetHash());
+
+    pow.getdouble() < GetKOSThreshold(bnTarget, k);
+}
```
Comment (Greg-Griffith) [inline src/bobtail/bobtail.cpp:115]: there is a return statement missing here

File: src/bobtail/bobtail.cpp [hunk 2]
```diff
@@ -107,3 +123,13 @@ unsigned int weakPOWfromPOW(unsigned int nBits) {
 
     return a.GetCompact();
 }
+
+double GetKOSThreshold(arith_uint256 target, uint8_t k)
```
Comment (Greg-Griffith) [inline src/bobtail/bobtail.cpp:127]: is it possible to change the math to ints? and then maybe move the decimal after?

File: src/bobtail/bobtail.cpp (hunk 1 above)
Comment (bissias) [reply src/bobtail/bobtail.cpp:115]: Fixed.

File: src/bobtail/bobtail.cpp (hunk 2 above)
Comment (bissias) [reply src/bobtail/bobtail.cpp:127]: I don't think so. I'm not aware of an integer type that won't overflow.

**Pull request merged**